lib/capistrano-deploy/hipchat.rb

```diff
@@ -5,23 +5,67 @@ def self.load_into(configuration)
         namespace :hipchat do
           require 'hipchat'
 
-          set(:deployer) { %x(git config user.name).chomp }
+          set(:deployer) { human }
+
           desc 'Set notification in hipchat for deployment start'
           task :start_msg do
             client = HipChat::Client.new(hipchat_token)
-            deploy_text = "#{deployer} is starting deploy of '#{app_name.upcase}' from branch '#{branch.upcase}' to #{current_stage.upcase}"
-
+            deploy_text = "#{human} is deploying #{deployment_name} to #{environment_string}."
             client[hipchat_room].send('Deploy', deploy_text, color: 'yellow')
           end
 
           desc 'Set notification in hipchat for deployment end'
           task :end_msg do
             client = HipChat::Client.new(hipchat_token)
-            deploy_text = "#{deployer} completed the deploy of '#{app_name.upcase}' from branch '#{branch.upcase}' to #{current_stage.upcase}"
+            deploy_text = "#{human} finished deploying #{deployment_name} to #{environment_string}."
 
             client[hipchat_room].send('Deploy', deploy_text, color: 'green')
           end
+
+          def human
+            user = ENV['HIPCHAT_USER'] || fetch(:hipchat_human, %x(git config user.name).chomp)
+            user = user || if (u = %x{git config user.name}.strip) != ''
+                             u
+                           elsif (u = ENV['USER']) != ''
+                             u
+                           else
+                             'Someone'
+                           end
+            user
+          end
+
+          def deployment_name
+             if fetch(:branch, nil)
+               branch = fetch(:branch)
+               name = "#{application_name}/#{branch}"
```

Comment:
You could probably do without the `name` variable (like the `environment_string` method).